test/python/transpiler/test_synthesis.py

```diff
@@ -140,8 +140,8 @@ def test_paper_example(self):
         c_compare.t(q[0])
         c_compare.cx(q[3], q[0])
         c_compare.t(q[0])
-        c_compare.cx(q[1], q[0])
         c_compare.cx(q[2], q[1])
+        c_compare.cx(q[1], q[0])
```

Comment:
This change drifts away from the example in the paper and it seems related with [this comment](https://github.com/Qiskit/qiskit-terra/pull/5044/files#r486469674).
<img width="384" alt="Screen Shot 2020-09-10 at 12 15 01 PM" src="https://user-images.githubusercontent.com/766693/92761521-c58cc080-f35f-11ea-9e9e-aa133cfb26a5.png">

Reply:
I checked the paper example again when the test was failing and realized it was wrong. The unitary in this case should be diagonal, but the circuit in the paper is not.